backend/src/services/identity-oidc-auth/identity-oidc-auth-service.ts

```diff
@@ -87,6 +87,28 @@ export const identityOidcAuthServiceFactory = ({
     if (!identity) throw new UnauthorizedError({ message: "Identity not found" });
 
     const org = await orgDAL.findById(identity.orgId);
+    const isSubOrgIdentity = Boolean(org.rootOrgId);
+
+    // If the identity is a sub-org identity, then the scope is always the org.id, and if it's a root org identity, then we need to resolve the scope if a subOrganizationName is specified
+    let subOrganizationId = isSubOrgIdentity ? org.id : null;
+
+    if (subOrganizationName) {
```

Comment:
Found a major issue. I think the sub-organization check should happen at last and then it should throw an error suborg not found.

The problem is - i had no clue the operation failed and the token was not scoped to sub-org. Thus my get project still returned root organization one